apps/balancer-tools/src/app/apr/round/(components)/PoolsCards.tsx

```diff
@@ -92,21 +82,10 @@ export async function PoolAPR({
   roundId?: string;
   poolId: string;
 }) {
-  // TODO: aggregate historical pool APR when roundId is not provided
+  // TODO BAL-646: aggregate historical pool APR when roundId is not provided
   if (!roundId) return null;
 
-  const round = Round.getRoundByNumber(roundId);
-
-  const pool = new Pool(poolId);
-
-  const [balPriceUSD, tvl, votingShare] = await Promise.all([
-    getBALPriceByRound(round),
-    // TODO: must select the correct network
-    BalancerAPI.getPoolTotalLiquidityUSD(pool.gauge?.network || 1, pool.id),
-    getPoolRelativeWeight(poolId, round.endDate.getTime() / 1000),
-  ]);
-
-  const apr = calculateRoundAPR(round, votingShare, tvl, balPriceUSD) * 100;
+  const { apr } = await calculateAPRForPool({ poolId, roundId });
```

Comment:
why complicate with object params? `#should` the same standard we've already established in the code:
```suggestion
  const apr = await calculateAPRForPool(poolId, roundId);
```

Reply:
because on other occasions we want to get the params that were used to calculate the APR. For example:

```
  const { apr, tvl, votingShare } = await calculateAPRForPool({
    poolId,
    roundId,
  });

```

This way, we don't have to fetch `votingShare` again. Maybe we can rename `calculateAPR` to something that is more obvious that you can get all the params

Reply:
just saw your comment 👍  we are aligned kkkk